packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -5677,44 +5672,44 @@ abstract class RenderObjectElement extends Element {
     }
 
     // Clean up any of the remaining middle nodes from the old list.
-    if (haveOldChildren && oldKeyedChildren.isNotEmpty) {
+    if (haveOldChildren && oldKeyedChildren!.isNotEmpty) {
       for (final Element oldChild in oldKeyedChildren.values) {
         if (forgottenChildren == null || !forgottenChildren.contains(oldChild))
           deactivateChild(oldChild);
       }
     }
 
-    return newChildren;
+    return newChildren.cast<Element>();
```

Comment:
we need to remove this cast (and not replace it with a copy). how we do that...

Reply:
the real solution is a dart feature to cast the list cheaply. But in the meantime, let's make a private Element subclass that we create a single instance of and use that as our placeholder, and just before we return let's verify that the list doesn't contain any of them.

Reply:
Done